auth/genericoauth/authority_verifier.go

```diff
@@ -0,0 +1,74 @@
+package genericoauth
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strings"
+
+	"golang.org/x/oauth2"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/atc/auth/verifier"
+)
+
+type AuthorityVerifier struct {
+	authority string
+}
+
+func NewAuthorityVerifier(
+	authority string,
+) verifier.Verifier {
+	return AuthorityVerifier{
+		authority: authority,
+	}
+}
+
+type GenericOAuthToken struct {
```

Comment:
I basically copied the scope verifier so you would be right about the code duplication. It seemed the speediest way to get something working. I can probably create a more generic verifier since the scope and authority checks are basically the same.